UOP1_Project/Assets/Scripts/CameraManager.cs

```diff
@@ -18,36 +23,63 @@ public class CameraManager : MonoBehaviour
 	[Tooltip("The CameraManager listens to this event, fired by objects in any scene, to adapt camera position")]
 	[SerializeField] private TransformEventChannelSO _frameObjectChannel = default;
 
+	[Tooltip("The CameraManager listens to this event, fired by objects in any scene, to swap the active virtual camera")]
+	[SerializeField] private VcamEventChannelSO _VcamEventChannel = default;
+
 
 	private bool _cameraMovementLock = false;
 
 	public void SetupProtagonistVirtualCamera(Transform target)
 	{
-		freeLookVCam.Follow = target;
-		freeLookVCam.LookAt = target;
+		foreach (CinemachineVirtualCameraBase vcam in vcamsInScene)
+		{
+			vcam.LookAt = target;
+			vcam.Follow = target;
+		}
```

Comment:
This centralised auto-populating list might break if for some reason the function is called from outside before OnEnable.

What about the cameras setup themselves? I know it's a bit against the idea of having the centralised manager, but it would work well.

Reply:
That method is part of a callback registered in onenable so I don't see a case where it can be called before on enable. But anyway I don't quite get what you mean by getting the cameras to set themselves up. Should I create a special vcam prefab with a script in it and do it there?

Reply:
'm not 100% sure how to do it. But for instance, if we do it your way then any Vcam will always follow and look at the player. This is not necessarily what we want. The beauty of Cinemachine is that we can create Vcams that are static, on rails, etc. and really tweak each shot in a different way.

So maybe what I'm thinking of is a script which we apply to the Vcams that we want to find the player. Maybe the script is a simple `VCamRequiresPlayerTarget` script with 2 booleans: looksAtPlayer and followsPlayer.

The `CameraManager` would do something like you did here, but look for the `VCamRequiresPlayerTarget` in the scene, and provide the player's target. These scripts would use it for their LookAt, Follow, for both, or for none, based on the booleans above.

How does it sound?

Reply:
sounds good. I will implement it.